src/random_test.cpp

```diff
@@ -283,35 +322,29 @@ int sum_of_all_server_options() {
   return total;
 }
 
-/* pick some algorith */
+/* pick some algorithm */
 inline static std::string pick_algorithm_lock() {
-  /* pick algorith for current sql */
-  static auto lock = opt_string(LOCK);
-  static auto algorithm = opt_string(ALGORITHM);
-  std::string locks[] = {"DEFAULT", "EXCLUSIVE", "SHARED", "NONE"};
-  std::string algorithms[] = {"INPLACE", "COPY", "INSTANT", "DEFAULT"};
+
   std::string current_lock;
   std::string current_algo;
-  if (lock.compare("all") == 0 && algorithm.compare("all") == 0) {
-    auto lock_index = rand_int(3);
-    auto algo_index = rand_int(3);
-    /* lock=none;algo=inplace|copy not supported */
-    if (lock_index == 3 && algo_index < 2)
-      lock_index = 0;
-    current_lock = locks[lock_index];
-    current_algo = algorithms[algo_index];
-  } else if (lock.compare("all") == 0) {
-    auto lock_index = rand_int(3);
-    current_lock = locks[lock_index];
-    current_algo = algorithm;
-  } else if (algorithm.compare("all") == 0) {
-    auto algo_index = rand_int(3);
-    current_algo = algorithms[algo_index];
-    current_lock = lock;
-  } else {
-    current_lock = lock;
-    current_algo = algorithm;
+
+  current_lock = locks[rand_int(locks.size() - 1)];
+
+/* With LOCK=NONE, ALGORITHM=INSTANT/COPY is not supported */
+  if (current_lock == "NONE") {
+    if (std::count(algorithms.begin(), algorithms.end(), "INSTANT"))
+      algorithms.erase(std::find(algorithms.begin(), algorithms.end(), "INSTANT"));
```

Comment:
it would remove INSTANT from algorithms? the remaining SQL will generate SQL using INSTANT

Reply:
Hi Rahul,
The function name is pick_algorithm_lock, , it returns both lock and algorithm together. So, whenever, LOCK=NONE, algorithm=INSTANT will not be supported. Hence, we don't need it

Reply:
you are modifying the global variable algorithms. once that entry is removed. It would not be available. See if there is any SQL using INSTANT.

Reply:
In pstress, we are not using ALGORITHM standalone in any generated SQL, it is always used together with LOCK.
For all lock modes, INSTANT is not supported

Reply:
for ex . pstress does not generate queries like these
`ALTER TABLE tbl_name DROP PRIMARY KEY, ALGORITHM=COPY;`

The way you have designed it is to return value like this
`return " LOCK=" + current_lock + ", ALGORITHM=" + current_algo;`

Reply:
IMO, this is a design issue. Ideally, we should have 2 different functions to pick lock and algorithm seperately such that we can also have queries which have LOCK and ALGORITHM independently.
for ex.
`ALTER TABLE tbl_name .. ALGORITHM=INSTANT;`
`ALTER TABLE tbl_name .. LOCK=NONE;`
There are several other issues like , there is no option to set the probability of --alter-lock or --alter-algorithm queries
If you agree with this, we can close this pull request. 
I request you to kindly fix it in a proper way.

Reply:
LOCK=DEFAULT or ALOGRITHM=DEFAULT will serve the purpose of using only one.  

ALTER TABLE tbl_name .. ALGORITHM=INSTANT; 
would be same as
ALTER TABLE tbl_name .. ALGORITHM=INSTANT , LOCK=DEFAULT;

If required we can open another jira ticket for --alter-lock-algo-prob.
this pull request is just to parse option provided by user.